infrastructure-playbooks/purge-cluster.yml

```diff
@@ -419,51 +340,61 @@
 
   - name: get ceph block partitions
     shell: |
-      blkid | awk '/ceph block"/ { sub (":", "", $1); print $1 }'
-    when: ceph_block_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block"
     failed_when: false
     register: ceph_block_partition_to_erase_path
 
   - name: get ceph journal partitions
     shell: |
-      blkid | awk '/ceph journal/ { sub (":", "", $1); print $1 }'
-    when: ceph_journal_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph journal"
     failed_when: false
     register: ceph_journal_partition_to_erase_path
 
   - name: get ceph db partitions
     shell: |
-      blkid | awk '/ceph block.db/ { sub (":", "", $1); print $1 }'
-    when: ceph_db_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block.db"
     failed_when: false
     register: ceph_db_partition_to_erase_path
 
   - name: get ceph wal partitions
     shell: |
-      blkid | awk '/ceph block.wal/ { sub (":", "", $1); print $1 }'
-    when: ceph_wal_partlabels.rc == 0
+      blkid -o device -t PARTLABEL="ceph block.wal"
     failed_when: false
     register: ceph_wal_partition_to_erase_path
 
+  - name: set_fact combined_devices_list
+    set_fact:
+      combined_devices_list: "{{ ceph_data_partition_to_erase_path.stdout_lines +
+                                 ceph_lockbox_partition_to_erase_path.stdout_lines +
+                                 ceph_block_partition_to_erase_path.stdout_lines +
+                                 ceph_journal_partition_to_erase_path.stdout_lines +
+                                 ceph_db_partition_to_erase_path.stdout_lines +
+                                 ceph_wal_partition_to_erase_path.stdout_lines }}"
+
+  - name: resolve parent device
+    shell: echo /dev/$(lsblk -no pkname "{{ item }}")
```

Comment:
do really need shell here?

Reply:
leftover

Reply:
actually, yes, we need `shell` to expand to `$()`